Sitefinity CLI/PackageManagement/NuGetApiClient.cs

```diff
@@ -135,16 +163,34 @@ private async Task<XDocument> GetPackageXmlDocument(string id, string version, I
             }
 
             HttpResponseMessage response = null;
-            foreach (string source in sources)
+            var v3Sources = sources.Where(x => x.Contains(Constants.ApiV3SourceString));
+            var v2Sources = sources.Except(v3Sources);
+            foreach (string source in v3Sources)
             {
-                string sourceUrl = source.TrimEnd('/');
-                response = await this.httpClient.GetAsync($"{sourceUrl}/Packages(Id='{id}',Version='{version}')");
+                string sourceUrl = this.GetBaseAddress(httpClient, source).Result.TrimEnd('/');
+                string loweredId = id.ToLowerInvariant();
+                response = await httpClient.GetAsync($"{sourceUrl}/{loweredId}/{version}/{loweredId}.nuspec");
                 if (response.StatusCode == HttpStatusCode.OK)
                 {
+                    this.isV2 = false;
                     break;
                 }
             }
 
+            if (isV2)
+            {
```

Comment:
extract into a method for better readability.

Reply:
the code is refactored